hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/TokenTupleUtils.java

```diff
@@ -150,25 +171,163 @@ private static List<Tuple> royaltyFeesTupleListFor(@NonNull final Token token) {
+                            .hasDenominatingTokenId();
                     final var denominatingTokenId = hasFallbackDenominatingTokenId
-                            ? fee.royaltyFee().fallbackFee().denominatingTokenId()
+                            ? fee.royaltyFeeOrElse(RoyaltyFee.DEFAULT)
+                                    .fallbackFee()
+                                    .denominatingTokenId()
                             : ZERO_TOKEN_ID;
                     return Tuple.of(
-                            fee.royaltyFee()
+                            fee.royaltyFeeOrElse(RoyaltyFee.DEFAULT)
                                     .exchangeValueFractionOrElse(ZERO_FRACTION)
                                     .numerator(),
-                            fee.royaltyFee()
+                            fee.royaltyFeeOrElse(RoyaltyFee.DEFAULT)
                                     .exchangeValueFractionOrElse(ZERO_FRACTION)
                                     .denominator(),
-                            fee.royaltyFee().fallbackFeeOrElse(ZERO_FIXED_FEE).amount(),
+                            fee.royaltyFeeOrElse(RoyaltyFee.DEFAULT)
+                                    .fallbackFeeOrElse(ZERO_FIXED_FEE)
+                                    .amount(),
                             headlongAddressOf(denominatingTokenId),
                             !hasFallbackDenominatingTokenId,
                             headlongAddressOf(fee.feeCollectorAccountIdOrElse(ZERO_ACCOUNT_ID)));
                 })
                 .toList();
     }
+
+    /**
+     * Returns a list of Tuples defined by the {@code TokenInfo} struct
+     * struct TokenInfo { HederaToken token; int64 totalSupply; bool deleted; bool defaultKycStatus; bool pauseStatus; FixedFee[] fixedFees; FractionalFee[] fractionalFees; RoyaltyFee[] royaltyFees; string ledgerId; }
+     * @param token the token to get the token info for
+     * @return Tuple encoding of the TokenInfo
+     */
+    @NonNull
+    public static Tuple tokenInfoTupleFor(@NonNull final Token token, @NonNull final String ledgerId) {
+        final var fixedFees = fixedFeesTupleListFor(token);
+        final var fractionalFees = fractionalFeesTupleListFor(token);
+        final var royaltyFees = royaltyFeesTupleListFor(token);
+
+        return Tuple.of(
+                hederaTokenTupleFor(token),
+                token.totalSupply(),
+                token.deleted(),
+                token.accountsKycGrantedByDefault(),
+                token.paused(),
+                fixedFees.toArray(new Tuple[fixedFees.size()]),
+                fractionalFees.toArray(new Tuple[fractionalFees.size()]),
+                royaltyFees.toArray(new Tuple[royaltyFees.size()]),
+                ledgerId);
+    }
+
+    /**
+     * Returns a tuple of the {@code FungibleTokenInfo} struct
+     * {@see https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L203 }
```

Comment:
Something's wrong with this JavaDoc too, and the next method:

![image](https://github.com/hashgraph/hedera-services/assets/117694041/93a113c0-3c71-4188-951f-3e58846cfbb2)

Normally of course I wouldn't even notice much less care but it happens to be catching my eye over in IntelliJ.